common/src/main/java/com/wynntils/wynn/model/PlayerInventoryModel.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model;
+
+import com.wynntils.core.managers.Model;
+import com.wynntils.mc.event.ContainerSetContentEvent;
+import com.wynntils.mc.event.SetSlotEvent;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wynn.event.WorldStateEvent;
+import com.wynntils.wynn.utils.ContainerUtils;
+import java.util.Objects;
+import net.minecraft.world.inventory.InventoryMenu;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class PlayerInventoryModel extends Model {
+
+    private static int emeralds = 0;
+    private static int openSlots = 0;
+
+    public static void init() {
+        resetCache();
+    }
+
+    public static void disable() {
+        resetCache();
+    }
+
+    @SubscribeEvent
+    public static void onWorldChange(WorldStateEvent e) {
+        if (e.getNewState() == WorldStateManager.State.WORLD) {
+            updateCache();
+        } else {
+            resetCache();
+        }
+    }
+
+    @SubscribeEvent
+    public static void onContainerSetEvent(ContainerSetContentEvent e) {
+        // Only update if the container is the player inventory
```

Comment:
Will this event be sent out "post" updating the inventory? So that `updateCache()` which grabs the players current inventory will really see the new content? (Same question about onSlotSetEvent below)

Reply:
Fixed!